nav2_smoother/src/simple_smoother.cpp

```diff
@@ -174,7 +176,7 @@ bool SimpleSmoother::smoothImpl(
           "Returning the last path before the infeasibility was introduced.");
         path = last_path;
         updateApproximatePathOrientations(path, reversing_segment);
-        return false;
+        throw nav2_core::SmoothedPathInCollision("Smoothed Path in collision");
```

Comment:
One thing I kind of liked about this was that we'd still try to smooth the other path segments. Just because one is in collision and we stop doesn't mean others couldn't be refined too.

As the warning mentions, this actually returns a still-valid path, just an iteration prior to the collision so its still valid. I'm not sure that throwing here would be the right move.

Reply:
Yeah, I agree :). That was thing I mentioned here: 

> From reading the server this weekend it seems like we still would like the server to succeed if it times out. (I.e partial smoothing has occurred) and just set a blackboard var to let anything else know that the smoothing was not completed.

I assumed partial smoothing is better than none at all :) but I wasn't sure.